src/model/resource_set.js

```diff
@@ -19,8 +19,130 @@
  */
 
 /**
- * @class ResourceSet is the top level interface for sets of parts of a project.
+ * @class ResourceSet is the top level interface for sets of parts of a project,
+ *     not including the project itself. Resource sets can only contain one type
+ *     of resource, and cannot contain multiple resources with the same name.
+ *
  */
  class ResourceSet {
-  // TODO: add methods, fields, etc.
+  /**
+   * ResourceSet Class.
+   * @param {string} resourceSetName The name for the resource set.
+   * @param {string} projectName The name of the project the resource belongs to.
+   * @param {Constructor} resourceConstructor The constructor for the type of
+   *     resource the set will manage.
+   * @constructor
+   */
+  constructor(resourceSetName, projectName, resourceConstructor) {
+    /**
+     * The name of the resource set.
+     * @type {string}
+     */
+     this.name = resourceSetName;
+    /**
+     * The name of the project the resource belongs to.
+     * @type {string}
+     */
+     this.project = projectName;
+    /**
+     * The constructor for type of resource the set manages.
+     * @type {Constructor}
+     */
+     this.resourceConstructor = resourceConstructor;
+  }
+
+  /**
+   * Writes the resource set to local storage.
+   */
+  saveSetToLocalStorage() {
```

Comment:
Let's implement the local storage bits at a higher level, perhaps by passing a storage mechanism into each save method, that has the means to assign strings (file contents) into named locations (filenames, or local storage). Then the Resource subclasses won't have to worry about how they are being stored.